Simbody/src/CableSpan.cpp

```diff
@@ -0,0 +1,3028 @@
+    {
+        realizePosition(state);
+        return Value<CableSpanData::Pos>::downcast(
+            getSubsystem().getCacheEntry(state, m_indexDataPos));
+    }
+
+    const CableSpanData::Vel& getDataVel(const State& state) const
+    {
+        realizeVelocity(state);
+        return Value<CableSpanData::Vel>::downcast(
+            getSubsystem().getCacheEntry(state, m_indexDataVel));
+    }
+
+    //--------------------------------------------------------------------------
+    // Accessors
+    //--------------------------------------------------------------------------
+
+    const CurveSegment& getObstacleCurveSegment(ObstacleIndex ix) const
+    {
+        return m_curveSegments[ix];
+    }
+
+    CurveSegment& updObstacleCurveSegment(ObstacleIndex ix)
+    {
+        return m_curveSegments[ix];
+    }
+
+    // Get the origin point in body fixed coordinates.
+    const Vec3& getOriginPoint_B() const
+    {
+        return m_originPoint_B;
+    }
+
+    // Get the termination point in body fixed coordinates.
+    const Vec3& getTerminationPoint_B() const
+    {
+        return m_terminationPoint_B;
+    }
+
+    CableSpanParameters& updParameters()
+    {
+        return m_parameters;
+    }
+
+    const CableSpanParameters& getParameters() const
+    {
+        return m_parameters;
+    }
+
+    //--------------------------------------------------------------------------
+    // Helper functions: End-points frame transformations.
+    //--------------------------------------------------------------------------
+
+    // Get body to which this cable's origin point is rigidly attached to.
+    const Mobod& getOriginBody() const
+    {
+        return getSubsystem()
+            .getMultibodySystem()
+            .getMatterSubsystem()
+            .getMobilizedBody(m_originBody);
+    }
+
+    // Get body to which this cable's termination point is rigidly attached to.
+    const Mobod& getTerminationBody() const
+    {
+        return getSubsystem()
+            .getMultibodySystem()
+            .getMatterSubsystem()
+            .getMobilizedBody(m_terminationBody);
+    }
+
+    Vec3 calcOriginPointInGround(const State& state) const
+    {
+        return getOriginBody().getBodyTransform(state).shiftFrameStationToBase(
+            m_originPoint_B);
+    }
+
+    Vec3 calcTerminationPointInGround(const State& state) const
+    {
+        return getTerminationBody()
+            .getBodyTransform(state)
+            .shiftFrameStationToBase(m_terminationPoint_B);
+    }
+
+    //--------------------------------------------------------------------------
+    // Helpers functions: Accessing obstacles in contact with cable.
+    //--------------------------------------------------------------------------
+
+    // Iterate over the cable's curve segments, and call provided function for
+    // those that are in contact with the obstacle's surfacce.
+    void forEachActiveCurveSegment(
```

Comment:
It might be that this helper function is actually more word-y than exposing iterators to the curve segments so that downstream callers can use their own algorithms or `for` range loops (e.g. much of your code for generating decorations doesn't use this, but does sometimes only want a filtered subset of data)